cms/static/js/spec/views/group_configuration_spec.js

```diff
@@ -626,4 +633,166 @@ define([
             });
         });
     });
+
+    describe('GroupList', function() {
+        var newGroupCss = '.new-button',
+            addGroupCss = '.action-add',
+            createGroups, renderView, verifyEditingGroup;
+
+        createGroups = function (groupNames) {
+            return new GroupCollection(_.map(groupNames, function (groupName) {
+                return {name: groupName};
+            }));
+        };
+
+        renderView = function(groupNames) {
+            var view = new GroupList({collection: createGroups(groupNames || [])}).render();
+            appendSetFixtures(view.el);
+            return view;
+        };
+
+        verifyEditingGroup = function(view) {
+            expect(view.$('.group-configuration-edit')).toExist();
+            expect(view.$(newGroupCss)).not.toExist();
+            expect(view.$(addGroupCss)).not.toExist();
+        };
+
+        beforeEach(function() {
+            TemplateHelpers.installTemplates(
+                ['add-list-item', 'no-content-groups', 'content-group-edit', 'content-group-details']
+            );
+        });
+
+        it('shows a message when no groups are present', function() {
+            expect(renderView().$('.no-group-configurations-content')).toExist();
+        });
+
+        it('can create an initial group', function() {
+            var view = renderView();
+            view.$(newGroupCss).click();
+            verifyEditingGroup(view);
+        });
+
+        it('can render groups', function() {
+            var groupNames = ['Group 1', 'Group 2', 'Group 3'];
+            renderView(groupNames).$('.group-configuration-details').each(function(index) {
+                expect($(this)).toContainText(groupNames[index]);
+            });
+        });
+
+        it('can add another group', function() {
+            var view = renderView(['Test Group']);
+            view.$(addGroupCss).click();
+            verifyEditingGroup(view);
+        });
+
+        it('only edits one form at a time', function() {
+            // Should prevent the user from opening more than one edit
+            // form at a time by removing the add button(s) when
+            // editing a group.
+            var view = renderView();
+            view.collection.add({name: 'Editing Group', editing: true});
+            verifyEditingGroup(view);
+        });
+    });
+
+    describe('GroupItem', function() {
```

Comment:
I'm happy to go that route. I wasn't quite sure where to draw the line, so I just decided to test every component.